network/proxy.md

```diff
@@ -6,68 +6,154 @@ keywords: network, networking, proxy, client
+
+When you run containers, proxy-related environment variables such as
+`HTTP_PROXY` reflect the values set in the configuration file.
+
+```console
+$ docker run -it alpine
+/ # echo $HTTP_PROXY
+http://192.168.1.12:3128
+```
+
+### Build with a proxy configuration
+
+When you invoke a build, proxy-related build arguments are pre-populated
+automatically, based on the proxy settings in your Docker client configuration
+file. You can access the proxy values using `ARG` instructions in your
+Dockerfile:
+
+```dockerfile
+FROM alpine
+ARG HTTPS_PROXY
+RUN apk update    # uses httpsProxy from ~/.docker/config.json
+```
+
+### Configure proxy settings per daemon
+
+The `proxies.default` key configures the proxy settings for all daemons that
+the client connects to. To configure the proxies for individual daemons,
+specify the proxy configuration under `proxies.<hostname>` in the client
+configuration file, where hostname is the address of the daemon.
+
+For example:
+
+```json
+{
+ "proxies": {
+   "https://docker-daemon1.example.com": {
+     "httpProxy": "http://192.168.1.12:3128",
+     "httpsProxy": "https://192.168.1.12:3129",
+     "noProxy": "*.test.example.com,.example.org,127.0.0.0/8"
+   }
+ }
+}
+```
+
+## Set proxy using the CLI
+
+Instead of [configuring the Docker client](#configure-the-docker-client),
+you can specify proxy configurations on the command-line when you invoke the
+`docker build` and `docker run` commands.
+
+Proxy configuration on the command-line uses the `--build-arg` flag for builds,
+and the `--env` flag for when you want to run containers with a proxy.
+
+```console
+$ docker build --build-arg HTTP_PROXY="http://192.168.1.12:3128" .
+$ docker run --env HTTP_PROXY="http://192.168.1.12:3128" redis
+```
+
+For a list of all the proxy-related build arguments that you can use with the
+`docker build` command, see
+[Predefined ARGs](../engine/reference/builder.md#predefined-args).
+These proxy values are only available in the build container.
+They're not included in the build output.
+
+## Proxy as environment variable for builds
+
+Don't use the `ENV` Dockerfile instruction to specify proxy settings for builds.
+Use build arguments instead.
+
+Using environment variables for proxies embeds the configuration into the image.
+If the proxy is an internal proxy, it might not be accessible for containers
+created from that image.
+
+Embedding proxy settings in images also poses a security risk, as the values
+may include sensitive information.
+
+## About proxy environment variables
+
+There's no standard that defines how web clients should handle proxy 
+environment variables, or the format for defining them. If you're interested
+in learning more about the history of these variables, check out this blog post
+on the subject, written by the GitLab team:
+[We need to talk: Can we standardize NO_PROXY?](https://about.gitlab.com/blog/2021/01/27/we-need-to-talk-no-proxy/){: target="_blank" rel="noopener"}.
```

Comment:
Should this be more at the start of the page? I also see that some part about the `noProxy` format (wildcard etc) was removed. Perhaps that's ok because there is no common standard 🤔 (just thinking if we can give some guidelines somehow)

Reply:
Good call, I'll try moving it up on the page. It's a nice callout to have as an intro.